app/main.py

```diff
@@ -1,3 +1,22 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    result_all = []
+
+    def inner(*args, **kqargs) -> Any:
+
+        results = {"name": func, "arg": args}
+        for result in result_all:
+            if result["name"] == results["name"] \
+                    and result["arg"] == results["arg"]:
+                print("Getting from cache")
+                return result["result"]
+
+        print("Calculating new result")
+        result_func = func(*args, **kqargs)
+        results["result"] = result_func
+        result_all.append(results)
+        return result_func
+
+    return inner
```

Comment:
It's very complicated. Instead of list use dict `result_all = {}` and then you can make check without loop just with `if args in result_all:`